### PR TITLE
Document ALTER MATERIALIZED VIEW SET PROPERTIES

### PR DESCRIPTION
**Pull request by jhlodin**
Document changes discussed in #10774

### CODE REVIEW

File: docs/src/main/sphinx/sql/alter-table.rst
Comment (sunningCGo) [inline docs/src/main/sphinx/sql/alter-table.rst:99]: Is the table name "people" or "users"?

File: docs/src/main/sphinx/sql/alter-table.rst
Comment (sunningCGo) [inline docs/src/main/sphinx/sql/alter-table.rst:103]: "people" or "users"?

**Comment by sunningCGo** [comment]
Will you be documenting the support of the `DEFAULT` keyword in the other six kinds of statements?

**Comment by hashhar** [comment]
@jhlodin Please ping me/tag me once this looks ready to merge. I unfortunately don't have enough context to review this - I'll defer to @sunningCGo.

**Comment by jhlodin** [comment]
> Will you be documenting the support of the `DEFAULT` keyword in the other six kinds of statements?

I can as a follow-up PR once this language gets approved and merged. The most important inclusion is the `ALTER MV SET PROPERTIES` documentation.

**Review by sunningCGo — APPROVED** [review]
Looks good to me. Please remember to document the support of `DEFAULT` in the other six kinds of statements as a follow-up.

**Review by mosabua — APPROVED** [review]

**Comment by jhlodin** [comment]
Tracked documenting the support of `DEFAULT` keyword in this issue: https://github.com/trinodb/trino/issues/10869

Can we merge this PR?

**Comment by sunningCGo** [comment]
I don't have permission to merge things into trino but I think @mosabua does.

**Comment by mosabua** [comment]
I wish @sunningCGo  .. can we get some help from @electrum @martint @hashhar @findepi maybe?

File: docs/src/main/sphinx/sql/alter-table.rst
Comment (findinpath) [inline docs/src/main/sphinx/sql/alter-table.rst:105]: Please do add also examples on how to deal with multiple properties:
```
ALTER TABLE a SET PROPERTIES foo=123, bar=456
```

Also do take into consideration  mentioning property names/values containing spaces.

```
ALTER TABLE a SET PROPERTIES " s p a c e "='bar'
```

**Comment by findinpath** [comment]
Please do use different commits for:
- set properties
- materialized view

**Comment by jhlodin** [comment]
> Please do use different commits for:
> 
>     * set properties
> 
>     * materialized view

@findinpath do you mean separate commits for `ALTER MATERALIZED VIEW SET PROPERTIES` and `ALTER TABLE SET PROPERTIES`?

**Comment by findinpath** [comment]
> @findinpath do you mean separate commits for ALTER MATERIALIZED VIEW SET PROPERTIES and ALTER TABLE SET PROPERTIES?
Yes.

**Comment by findinpath** [comment]
@sunningCGo I'm having a hard time to find a connector which supports the statement of altering the properties for materialized views.

Could you please document a simple test scenario for the functionality of dealing with this functionality?

**Review by sopel39 — APPROVED** [review]
lgtm % mind other comments

File: docs/src/main/sphinx/sql/alter-materialized-view.rst
Comment (findinpath) [inline docs/src/main/sphinx/sql/alter-materialized-view.rst:30]: `Excluding a property` -> `Omitting an already set property of the materialized view`

**Comment by sunningCGo** [comment]
> @sunningCGo I'm having a hard time to find a connector which supports the statement of altering the properties for materialized views.

Support in Starburst Hive Connector is being added in https://github.com/starburstdata/starburst-enterprise/pull/4523.

> Could you please document a simple test scenario for the functionality of dealing with this functionality?

There are some tests in https://github.com/starburstdata/starburst-enterprise/blob/2be54560b5dd8c1ccdc15fb72d5ba36d2d48a1ed/lib/starburst-redirection/src/test/java/com/starburstdata/presto/redirection/AbstractStarburstMaterializedViewsTest.java#L1165

> Besides io.trino.plugin.clickhouse.ClickHouseClient#setTableProperties (Clickhouse connector) is there any other connector at the moment supporting setting table properties?

No.

File: docs/src/main/sphinx/sql/alter-materialized-view.rst
Comment (hashhar) [inline docs/src/main/sphinx/sql/alter-materialized-view.rst:34]: Do defaults exist for every property @sunningCGo ? What happens if there is a null default but nulls are not allowed for a property?

File: docs/src/main/sphinx/sql/alter-materialized-view.rst
Comment (hashhar) [inline docs/src/main/sphinx/sql/alter-materialized-view.rst:53]: Do we have some mechanism (a system table like `system.metadata.table_properties`?) to inspect available properties and their data types and default values?

File: docs/src/main/sphinx/sql/alter-materialized-view.rst
Comment (joshthoward) [reply docs/src/main/sphinx/sql/alter-materialized-view.rst:34]: @hashhar there are tests for this behavior in `TestSetPropertiesTask.java`.

File: docs/src/main/sphinx/sql/alter-materialized-view.rst
Comment (martint) [inline docs/src/main/sphinx/sql/alter-materialized-view.rst:58]: Typo: `peopl` -> `people`

Comment (jhlodin) [reply docs/src/main/sphinx/sql/alter-materialized-view.rst]: Fixed.

File: docs/src/main/sphinx/sql/alter-materialized-view.rst
Comment (hashhar) [reply docs/src/main/sphinx/sql/alter-materialized-view.rst:53]: https://github.com/trinodb/trino/issues/10916

File: docs/src/main/sphinx/connector/alter-mv-set-properties-unsupported.fragment
Comment (hashhar) [inline docs/src/main/sphinx/connector/alter-mv-set-properties-unsupported.fragment:6]: Remove the `Other` part since once we have multiple such fragments (which we'll do once more syntax is added) are added this statement will become false. (And it would not be caught in review since it's a fragment).

Comment (jhlodin) [reply docs/src/main/sphinx/connector/alter-mv-set-properties-unsupported.fragment]: Done

**Review by hashhar — APPROVED** [review]

**Pull request merged**
